csrc/logical_domain_map.cpp

```diff
@@ -284,19 +284,25 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseLogicalDomainMap::map(
     bool consumer_has_e = consumer_tv_->sameAs(op->grad_query()) ||
         consumer_tv_->sameAs(op->grad_key());
 
+    size_t num_device_dim =
+        producer_logical.size() > 0 && producer_logical.at(0)->isDeviceDim()
+        ? 1
+        : 0;
```

Comment:
```suggestion
    NVF_ERROR(producer_logical.size() > 0);
    size_t num_device_dim = producer_logical.at(0)->isDeviceDim();
```

Reply:
The producer_size can be 0, for `philox_seed/offset` which are scalar tensors.
We just skip mapping for them, it is not an error.